webui/react/src/components/kit/Icon.tsx

```diff
@@ -204,21 +204,26 @@ const FourSquaresIcon: React.FC = () => (
 
 export type IconName = (typeof IconNameArray)[number];
 
-export interface Props {
-  color?: 'cancel' | 'error' | 'success';
-  name: IconName;
-  showTooltip?: boolean;
-  size?: IconSize;
-  title: string;
-}
-
-const Icon: React.FC<Props> = ({
-  name,
-  showTooltip = false,
-  size = 'medium',
-  title,
-  color,
-}: Props) => {
+export type Props =
+  | {
+      color?: 'cancel' | 'error' | 'success';
+      name: IconName;
+      showTooltip?: boolean;
+      size?: IconSize;
+      title: string;
+    }
+  | {
+      color?: 'cancel' | 'error' | 'success';
+      name: IconName;
+      size?: IconSize;
+      decorative: true;
+    };
```

Comment:
Just fooling around with this but couldn't we do something like
```suggestion
type CommonProps = {
      color?: 'cancel' | 'error' | 'success';
      name: IconName;
      size?: IconSize;
}
export type Props = CommonProps & (
  | {
      showTooltip?: boolean;
      title: string;
      decorative?: never
    }
  | {
      decorative: true;
    }
);
```
?

Reply:
Literally what I tried and when using `Icon` it would complain that `title` should be `undefined`.

Reply:
It seems like that approach only works when all of the differing properties are optional.

Reply:
i ran this locally and i only saw that type issue in the test code, which throws because we're spreading a `Props` object into an object that already has `title` defined.

tbh, the deduped version is slightly longer than the duped version, so I'm not super partial to it, but could you confirm?

Reply:
I tried copy/pasting your diff and I see:
```console
 ERROR(TypeScript)  Type '{ name: "star"; showTooltip: true; title: string; } | { color?: "error" | "cancel" | "success" | undefined; name: "home" | "dai-logo" | "arrow-left" | "arrow-right" | "add-small" | "close-small" | ... 76 more ... | "four-squares"; size?: "tiny" | ... 10 more ... | undefined; showTooltip: boolean; title: string; deco...' is not assignable to type 'IntrinsicAttributes & Props'.
  Type '{ color?: "error" | "cancel" | "success" | undefined; name: "home" | "dai-logo" | "arrow-left" | "arrow-right" | "add-small" | "close-small" | "search" | "arrow-down" | "arrow-up" | ... 73 more ... | "four-squares"; size?: "tiny" | ... 10 more ... | undefined; decorative: true; showTooltip: true; title: string; }' is not assignable to type 'IntrinsicAttributes & Props'.
    Property 'showTooltip' does not exist on type 'IntrinsicAttributes & CommonProps & { decorative: true; }'.
 FILE  /Users/claire/Programming/determined/webui/react/src/components/kit/Icon.test.tsx:20:41
```

Reply:
I suppose I can just rewrite that test. Didn't look too closely but I guess that test is just not correctly written? Normal props spread works just fine, it's just this weird case that doesn't work.

Reply:
Done, thanks for the help.

Reply:
Yeah -- the `props` variable is the same type of the `Icon` props, so the resulting object would potentially be invalid. If I passed `{ name: "filter", decorative: true }` into the setup function, the resulting object would have both `title` and `decorative` defined, which should fail.